snuba/cli/jobs.py

```diff
@@ -1,28 +1,54 @@
-from typing import Tuple
+from typing import Any, MutableMapping, Tuple
 
 import click
 
+from snuba.manual_jobs import JobSpec
 from snuba.manual_jobs.job_loader import JobLoader
+from snuba.manual_jobs.manifest_reader import ManifestReader
+
+JOB_SPECIFICATION_ERROR_MSG = "Missing job type and/or job id"
 
 
 @click.group()
 def jobs() -> None:
     pass
 
 
+def _parse_params(pairs: Tuple[str, ...]) -> MutableMapping[Any, Any]:
+    params = {}
+    for pair in pairs:
+        k, v = pair.split("=")
+        params[k] = v
+    return params
+
+
 @jobs.command()
-@click.argument("job_name")
+@click.option("--json_manifest", required=True)
+@click.option("--job_id")
 @click.option(
     "--dry_run",
     default=True,
 )
-@click.argument("pairs", nargs=-1)
-def run(*, job_name: str, dry_run: bool, pairs: Tuple[str, ...]) -> None:
+def run_from_manifest(*, json_manifest: str, job_id: str, dry_run: bool) -> None:
+    job_specs = ManifestReader.read(json_manifest)
+    for job_spec in job_specs:
```

Comment:
Maybe ManifestReader.read can return a map of job id to job spec

Reply:
I'd be fine with that